common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressOptions.java

```diff
@@ -144,6 +144,6 @@ public List<String> split(String value) {
     }
 
     public enum HttpClientType {
-        NETTY, OKHTTP
+        NETTY, OKHTTP, JDK
```

Comment:
_Can be done in this PR or pushed to the future._

We should change this from being a magical string to being the full class name of the HttpClientProvider used to create the HttpClient instance. Or, at least give this enum a field for that value (`NETTY("com.azure.core.http.netty.NettyAsyncHttpClientProvider")`).

That way we can use the logic in our service provider loader to get the explicit instance we want instead of needing to use the `switch` in `ApiPerfTestBase` and should allow for us to make the code more generic and not need updating every time we add a new HttpClient.

Reply:
Been thinking about it and in manual cmd line it's easier to write `--http-client jdk`, but what do you think about the new impl:
- known clients have friendly names (`netty`, `okhttp`, `vertx`, `jdk`). 
- we can also (in most cases) create instances that trust all certs
- we can create them without SPI (and don't need shading plugins, etc)
-  plus any full priovider name is supported if resolved with SPI

Tested it on core perf tests, works nicely:

`java -jar .\target\azure-core-perf-1.0.0-beta.1-jar-with-dependencies.jar binarydatareceive -w 0 -d 1 -p 1 --http-client com.azure.core.http.okhttp.OkHttpAsyncClientProvider  ...`

`java -jar .\target\azure-core-perf-1.0.0-beta.1-jar-with-dependencies.jar binarydatareceive -w 0 -d 1 -p 1 --http-client okhttp  ...`